### PR TITLE
Offspec spells won't work either.

### PR DESCRIPTION
**Pull request by cont1nuity**
This covers unlearned spells from off-specs which appear in the spellbook.

### CODE REVIEW

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua [hunk 1]
```diff
@@ -90,9 +90,9 @@ local GetSpellBookItemInfo = _G.GetSpellBookItemInfo or function(index, spellBan
     spellBank = (spellBank == "spell") and Enum.SpellBookSpellBank.Player or Enum.SpellBookSpellBank.Pet;
   end
   local info = C_SpellBook.GetSpellBookItemInfo(index, spellBank)
-  -- we are looking for "Spell" here, as "FutureSpell" and passives are not working with C_Spell.IsSpellInRange
-  if info and not info.isPassive and info.itemType == Enum.SpellBookItemType.Spell then
-    return info.itemType, info.spellID
+  -- we are looking for main spec "Spell" here, as "FutureSpell" and passives are not working with C_Spell.IsSpellInRange
+  if info and not info.isPassive and info.itemType == Enum.SpellBookItemType.Spell and not info.isOffSpec then
+    return info.itemType, info.spellID, info
```
Comment (InfusOnWoW) [inline LibRangeCheck-3.0/LibRangeCheck-3.0.lua:95]: So I dislike that we are checking lots of things inside the compability GetSpellBookItemInfo function. Imho ideally, the function should be a as close as possible to the behaviour of classic and retail. 

That is I think, the checking for isOffSpec, isPassive shouldn't be done in the compability function, but in the place where it is used.

Though I'm okay with merging the code as is.

**Review by InfusOnWoW — APPROVED** [review]

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua (hunk 1 above)
Comment (cont1nuity) [reply LibRangeCheck-3.0/LibRangeCheck-3.0.lua:95]: I agree and can rework that later today.

Comment (cont1nuity) [reply LibRangeCheck-3.0/LibRangeCheck-3.0.lua]: I updated the code to move the filtering out.

File: LibRangeCheck-3.0/LibRangeCheck-3.0.lua [hunk 2]
```diff
@@ -654,11 +655,13 @@ local function findSpellIdx(spellName)
   for i = 1, getNumSpells() do
     local spell = GetSpellBookItemName(i, BOOKTYPE_SPELL)
     if spell == spellName then
-      local spellType, spellID = GetSpellBookItemInfo(i, BOOKTYPE_SPELL)
-      if spellType == "SPELL" then -- classic/era
+      local spellType, spellID, spellInfo = GetSpellBookItemInfo(i, BOOKTYPE_SPELL)
+      if spellInfo then -- new API output available
+        if Enum.SpellBookItemType and spellInfo.itemType == Enum.SpellBookItemType.Spell and not spellInfo.isOffSpec then -- retail - filter for only active spec "SPELL"
```
[the remark this answers is not in the document]
Comment (cont1nuity) [reply LibRangeCheck-3.0/LibRangeCheck-3.0.lua:660]: I don't think so, as we are explicitly checking for the proper spell name before this. I am not aware of any of the range check spells being passive or having passives with the same name.
For good measure, I could re-add it, though, to make sure.

**Comment by InfusOnWoW** [comment]
Ah it's fine. Thanks again.

**Comment by cont1nuity** [comment]
A pleasure.

Until we run into the next change :)

**Pull request merged**